controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -90,11 +90,23 @@ func (r *DSCInitializationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	case len(instances.Items) == 1:
 		instance = &instances.Items[0]
 	case len(instances.Items) > 1:
-		message := fmt.Sprintf("only one instance of DSCInitialization object is allowed. Update existing instance name %s", req.Name)
-		_, _ = r.updateStatus(ctx, &instances.Items[0], func(saved *dsciv1.DSCInitialization) {
-			status.SetErrorCondition(&saved.Status.Conditions, status.DuplicateDSCInitialization, message)
-			saved.Status.Phase = status.PhaseError
-		})
+		// find out the one by created timestamp and use it as the default one
+		earliestDSCI := &instances.Items[0]
+		for _, instance := range instances.Items {
+			currentDSCI := instance
+			if currentDSCI.CreationTimestamp.Before(&earliestDSCI.CreationTimestamp) {
+				earliestDSCI = &currentDSCI
+			}
+		}
+		message := fmt.Sprintf("only one instance of DSCInitialization object is allowed. Please delete other instances than %s", earliestDSCI.Name)
+		// update all instances Message and Status
+		for _, instance := range instances.Items {
+			_, _ = r.updateStatus(ctx, &instance, func(saved *dsciv1.DSCInitialization) {
+				status.SetErrorCondition(&saved.Status.Conditions, status.DuplicateDSCInitialization, message)
+				saved.Status.Phase = status.PhaseError
+			})
+		}
```

Comment:
Why are we marking all of them as failed? That would make previously correctly configured DSCI marked as failed too. I think the rule you set is to not allow more than 1 being processed, so shouldn't we simply ignore new ones?

Reply:
good idea. i did not even think about this is gonna impact the first "good" instance.